.pre-commit-config.yaml

```diff
@@ -0,0 +1,11 @@
+repos:
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+      - id: fmt
+        args: ['--manifest-path', 'src/Simulation/qdk_sim_rs/Cargo.toml', '--']
+      - id: cargo-check
+        args: ['--manifest-path', 'src/Simulation/qdk_sim_rs/Cargo.toml', '--']
+      - id: clippy
```

Comment:
I'm actually not familiar with clippy, any links you'd recommend to get more info on it?

Reply:
Ah, yes, let me add that link... thanks for calling that out! In the meantime, https://github.com/rust-lang/rust-clippy and the list of lints checked by clippy at https://rust-lang.github.io/rust-clippy/master/index.html should be helpful.